[master] fix ephemeral interface addition

avances123 opened the pull request:
### What does this PR do?
Fixes KeyError exception when a not know interface appears

### What issues does this PR fix or reference?
Fixes: test if new interface exists in LAST_DATA if not, creates an empty set to make the diff

### Previous Behavior
exception in beacon when new interface appears 

### New Behavior
beacon executes flawlessly

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

avances123 committed:
fix ephemeral interface addition

twangboy reviewed (changes requested):
Please create a changelog and a test for this.

dwoz committed:
Add changelog and test for ephemeral interface KeyError fix

Add changelog fragment 64584.fixed.md and unit test
test_ephemeral_interface_no_keyerror() covering the case where a network
interface appears after LAST_STATS initialization, verifying no KeyError
is raised and the new interface fires a beacon event.
